backend/app/Http/Controllers/MarketplaceController.php

```diff
@@ -54,6 +55,33 @@ public function updateCard($card_id)
         return $this->getCardDetail($card_id);
     }
 
+    public function putTransaction($card_id)
+    {
+        //update card user's id with current id
+        $user = auth()->user();
+        $card = Card::find($card_id);
+        $cardList = Listing::find($card_id);
```

Comment:
This will find a listing with listing_id = the card_id. This will not return the correct record.